---
title: "paintor-subsetting"
author: "Paul Hook"
date: "2/1/2019"
output: html_document
---

```{r, loading-packages}
library(here)
library(tidyverse)
library(reshape2)
library(patchwork)
library(Gviz)
library(org.Hs.eg.db)
library(motifbreakR)
library(BSgenome.Hsapiens.UCSC.hg19)
library(SNPlocs.Hsapiens.dbSNP144.GRCh37)
```

```{r, reading-in-data}
# loading
null.path <- here::here("final.null.results.txt")
anno.path <- here::here("final.anno.results.txt")
overlap.path <- here::here("final.overlap.txt")

# loading and cleaning
null.mat <- readr::read_delim(null.path,delim = " ")

anno.mat <- readr::read_delim(anno.path,delim = " ")
length(unique(anno.mat$rsid))

overlap.mat <- readr::read_delim(overlap.path,delim = " ") %>%
  dplyr::select(-chr,-pos,-A1,-A2,-Zscore,-r.squared,-`-log10(P)`) %>%
  dplyr::mutate(all.sum=rowSums(.[4:30]),
                enrich.sum=rowSums(.[c(11:14,16:23,30)]))

# merging
fm.df <- merge(x=null.mat,y=anno.mat,by=names(null.mat)[1:10]) %>% 
  as_tibble() %>%
  dplyr::rename("PIP_null" = Posterior_Prob.x, "PIP_anno" = Posterior_Prob.y)

# make rsid/lead.snp table
snps <- fm.df %>%
  dplyr::select(rsid,lead.snp) %>%
  dplyr::group_by(rsid) %>%
  summarise(lead.snps=paste(lead.snp,collapse = ","))

# All output for Table S8
print.fm.df <- fm.df %>%
  dplyr::select(id, everything())
write_tsv(print.fm.df,"Table.S8_fm-results.txt")

# Merge with binary overlap file
overlap.df <- merge(x=fm.df,y=overlap.mat,by=c("rsid","id","lead.snp")) %>% 
  as_tibble()
write.table(overlap.df,file = here::here("finemap_overlap_results.txt"),quote = FALSE,sep = "\t",row.names = FALSE,col.names = TRUE)

# summarizing
fm.sum <- overlap.df %>%
  dplyr::group_by(lead.snp) %>%
  dplyr::summarise(total.snps=n(),
                   `> 0.1`=sum(PIP_null >= 0.1 | PIP_anno >= 0.1),
                   `> 0.5`=sum(PIP_null >= 0.5 | PIP_anno >= 0.5),
                   `> 0.9`=sum(PIP_null >= 0.9 | PIP_anno >= 0.9),
                   overlap.ten.enrich=sum((PIP_null >= 0.1 | PIP_anno >= 0.1) & enrich.sum >=1),
                   overlap.fifty.enrich=sum((PIP_null >= 0.5 | PIP_anno >= 0.5) & enrich.sum >=1),
                   overlap.ninety.enrich=sum((PIP_null >= 0.9 | PIP_anno >= 0.9) & enrich.sum >=1),
                   total.pp.null=round(sum(PIP_null),1), #adding posterior probs was mentioned in a PAINTOR paper
                   total.pp.annotation=round(sum(PIP_anno),1))

# How many loci have SNPs that reach these thresholds
summary(fm.sum$`> 0.1` > 0)
summary(fm.sum$`> 0.5` > 0)
summary(fm.sum$`> 0.9` > 0)

write.table(fm.sum,file = here::here("finemap_overlap_summary.txt"),quote = FALSE,sep = "\t",row.names = FALSE,col.names = TRUE)

# Plotting PIP with and without annotation
p <- ggplot(data=fm.df,aes(x=PIP_null,y=PIP_anno))
q <- p + geom_point() +
  ggtitle("Comparing posterior probabililty when performing finemapping\nwithout and with significant SZ annotation") +
  ylab("Annotation Posterior Probability (PIP_anno)") +
  xlab("Posterior Probability (PIP_null)") +
  theme_classic()
q
ggsave(q,filename = "posterior_probs.png",width = 7,height = 7,dpi="retina")

# cor
cor(fm.df$PIP_null,fm.df$PIP_anno,method = "spearman") 
```

Look at SNPs that were fine-mapped in multiple loci
```{r}
#Duplicated -- probably will need to address this and talk about how it can lead to additional insights
idx <- duplicated(fm.df$rsid) | duplicated(fm.df$rsid, fromLast = TRUE) 
summary(idx)
dup <- fm.df[idx, ] %>% dplyr::arrange(rsid)

length(unique(dup$rsid))
length(unique(dup$lead.snp))

dup.tmp <- dup %>%
  dplyr::filter((PIP_null >= 0.1 | PIP_anno >= 0.1))

idx <- duplicated(dup.tmp$rsid) | duplicated(dup.tmp$rsid, fromLast = TRUE) 
summary(idx)
dup.10 <- dup.tmp[idx, ] %>% dplyr::arrange(rsid) %>% dplyr::select(id,everything())
length(unique(dup.10$lead.snp))
length(unique(dup.10$rsid))

write_tsv(dup.10,path = "Table.S12_duplicated-snps-high-pip.txt")
```

```{r}
# subset rs4144797 locus
locus <- overlap.df %>%
  dplyr::filter(lead.snp == "rs4144797") %>% as_tibble()

plot1 <- ggplot() +
  geom_point(data = locus, aes(x = pos, y = r.squared), size = 2) +
  theme_classic(base_size = 16) +
  labs(x = "", y = bquote("R"^2)) +
  scale_y_continuous(limits = c(0,1)) +
  theme(axis.text.x = element_blank()) +
  scale_x_continuous(limits = c(min(locus$pos)-2000, max(locus$pos) + 2000))

plot2 <- ggplot() +
  geom_point(data = locus, aes(x = pos, y =`-log10(P)`), size = 2) +
  theme_classic(base_size = 16) +
  labs(x = "", y = bquote("-log10(P-values)")) +
  #scale_y_continuous(limits = c(0,1))
  theme(axis.text.x = element_blank()) +
  scale_x_continuous(limits = c(min(locus$pos)-2000, max(locus$pos) + 2000)) +
  #geom_text_repel(aes(pos, logP, label = rsid), data = locus[locus$rsid %in% #c("rs181813160","rs4144797"),],point.padding = 0.1,nudge_x = 0.5) +
  NULL

plot3 <- ggplot() +
  geom_point(data = locus, aes(x = pos, y = PIP_anno), size = 2) +
  theme_classic(base_size = 16) +
  labs(x = "Chr. 2 position", y = bquote("Finemapping PIP")) +
  scale_y_continuous(limits = c(0,1.1)) +
  geom_hline(yintercept = 0.1) +
  scale_x_continuous(limits = c(min(locus$pos)-2000, max(locus$pos) + 2000)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_text(aes(pos, PIP_null, label = rsid), data = locus[locus$rsid %in% c("rs181813160","rs4144797"),],nudge_y = 0.18)

total.plot <- plot1/plot2/plot3
ggsave(filename = "ngef.png",plot = total.plot,width = 6,height = 10)


```

```{r}
#Making a table for Mike Beer
beer.table <- locus %>%
  dplyr::select(chr,pos,id,rsid) %>%
  dplyr::mutate(svm=if_else(rsid == "rs181813160",true = "yes",false = "no"),
                genome="hg19") %>%
  dplyr::select(chr,pos,genome,id,rsid,svm)

write.table(beer.table, here::here("mike.beer","rs4144797_locus.txt"),quote = FALSE,sep = "\t",row.names = FALSE)
```

```{r}
# Chunking out only the subset of variants with >= 0.1 and overlap wtih enriched cells
subset <- overlap.df %>%
  dplyr::filter((PIP_null >= 0.1 | PIP_anno >= 0.1) & merged.anno.bed == 1) %>%
  dplyr::select(-merged.anno.bed) %>%
  dplyr::select(id,chr,pos,rsid,A1,A2,Zscore,lead.snp,everything())

write_tsv(subset,"Table.S9_finemap-causal-overlap.txt")
  
# enrich.sum will need to change slightly as it will need to include the other 2 annotations originally left out.
length(unique(subset$rsid))
length(unique(subset$lead.snp))


write.table(subset,file="Table.finemapping-overlap-subset.txt",quote = FALSE,sep = "\t",row.names = FALSE,col.names = TRUE)


```

```{r}
subset.table <- overlap.df %>%
  dplyr::filter(PIP_null >= 0.1 | PIP_anno >= 0.1) %>%
  dplyr::select(-id,-Zscore,-r.squared,-`-log10(P)`,-merged.anno.bed,-enrich.sum,-all.sum)

subset.table$`Cell Populations` <- apply(subset, 1, function(x) paste0(names(subset)[x == 1], collapse = ", "))

table <- subset.table %>%
  dplyr::select("RSID"=rsid,
                "Lead SNP"=lead.snp,
                "PIP with no annotation"=PIP_null,
                "PIP with annotation"=PIP_anno,`Cell Populations`) %>%
  dplyr::mutate(`Cell Populations`=if_else(`Cell Populations`=="","None",`Cell Populations`)) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"_sc","*")) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"_"," ")) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"Cones blue","Cones (blue)")) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"Cones green","Cones (green)")) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"Midbrain DA","Embryonic DA Midbrain")) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"Forebrain DA","Embryonic DA Forebrain")) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"CD4 T-cells mouse","CD4 T-cells")) %>%
  dplyr::mutate(`Cell Populations`=str_replace_all(`Cell Populations`,"CD8 T-cells mouse","CD8 T-cells"))

# Now just subset it out for NGEF, PDE4, CACNA1C
paper.loci <- table %>%
  dplyr::filter(`Lead SNP` == "rs4144797" | `Lead SNP` == "rs12129719" | `Lead SNP` == "rs2007044") %>%
  dplyr::arrange(`Lead SNP`,desc(`PIP with annotation`))

write_tsv(paper.loci, path = "Table.1_three-loci_finemap-snps.txt")
```

```{r}
snps <- as.character(unique(subset$rsid))
variants <- snps.from.rsid(rsid = snps,
                           dbSNP = SNPlocs.Hsapiens.dbSNP144.GRCh37,
                           search.genome = BSgenome.Hsapiens.UCSC.hg19)

results <- motifbreakR(snpList = variants,
                       filterp = TRUE,
                       pwmList = encodemotif,
                       threshold = 1e-4,
                       method = "ic",
                       bkg = c(A=0.25, C=0.25, G=0.25, T=0.25),
                       BPPARAM = BiocParallel::bpparam())

# Writing results into regular data frames
results.df <- data.frame(results)
variants.df <- data.frame(variants)

# Saving variants and results as .Rds this was run with "encodemotif" as the pwmList
saveRDS(variants,file = "motifVariants.rds")
saveRDS(results,file = "motifResults.rds")

variants <- readRDS("motifVariants.rds")
results <- readRDS("motifResults.rds")

# Writing results into regular data frames
results.df <- data.frame(results)
variants.df <- data.frame(variants)

# Merging results
mod <- variants.df %>% dplyr::select(seqnames,REF,ALT,start,SNP_id)
test <- merge(x=results.df, y=mod, by.x=c('seqnames','REF','ALT','snpPos'), by.y=c('seqnames','REF','ALT','start'))
# checking number of snps
length(unique(test$SNP_id))

# Making a readable SNP table
tf.table <- test %>%
  dplyr::select("rsid"=SNP_id,effect,geneSymbol,dataSource,providerName,"providerID"=providerId,seqMatch)
length(unique(tf.table$rsid))

tf.merge <- merge(x=tf.table,y=snps,by="rsid") %>%
  dplyr::select(rsid,lead.snps,everything())

write_tsv(tf.merge, "Table.S10_motif-results.txt")
  
# Checking on TFs and creating a list for TFs
tf.list <- test %>%
  dplyr::select(SNP_id,geneSymbol) %>%
  unique()
length(unique(tf.list$geneSymbol)) #335 unique TFs

tf.df <- as.data.frame.table(table(tf.list$geneSymbol)) %>%
  dplyr::arrange(-Freq) %>%
  dplyr::rename("TF.gene"=Var1)

p<-ggplot(data=tf.df, aes(x=reorder(TF.gene,-Freq), y=Freq)) +
  geom_bar(stat="identity") +
  theme(axis.text.x = element_blank())
p

write_tsv(tf.df,path = here::here("Table.S11_tf-freq.txt"))


# Making a plot
dat <- fm.df %>%
  dplyr::filter(rsid %in% unique(test$SNP_id))

egr1 <- test %>%
  dplyr::filter(geneSymbol == "EGR1") %>%
  dplyr::pull(SNP_id) %>% unique()

dat.egr1 <- fm.df %>%
  dplyr::filter(rsid %in% egr1)

dat.results <- results[names(results) == "rs181813160"]
dat.df <- data.frame(dat.results)
dat.results.strong <- dat.results[dat.results$effect == "strong"]
length(unique(dat.results.strong$geneSymbol))

ngef.results <- dat.results[(elementMetadata(dat.results)[, "providerName"] %in% c("EGR1_known9","EGR3_known_ref3","EGR4_known_ref2"))]

plotMB(ngef.results, "rs181813160",reverseMotif = TRUE, effect = "strong")
pdf(file = "rs181813160-egr-plot.pdf",width = 5,height = 6,bg="white")
plotMB(ngef.results, "rs181813160",reverseMotif = TRUE, effect = "strong")
dev.off()

# Making another plot
dat.results <- results[names(results) == "rs4144797"]
dat.df <- data.frame(dat.results)
dat.results.strong <- dat.results[dat.results$effect == "strong"]
ngef2.results <- data.frame(dat.results.strong)
length(unique(dat.results.strong$geneSymbol))

ngef.results.2 <- dat.results.strong[(elementMetadata(dat.results.strong)[, "providerName"] %in% "EGR1_disc3"),]

plotMB(ngef.results.2, "rs4144797",reverseMotif = TRUE, effect = "strong")
pdf(file = "rs4144797-plot.pdf",width = 5,height = 6,bg="white")
plotMB(ngef.results.2, "rs4144797",reverseMotif = TRUE, effect = "strong")
dev.off()
```

Making plots for rs181813160
```{r}
chr <- 'chr2'
gen <- 'hg19'

 
gtrack <- GenomeAxisTrack()
itrack <- IdeogramTrack(genome = gen, chromosome = chr,fontfamily="Helvetica",fontcolor="black",fontface=2,fontsize=12)

plotTracks(list(gtrack,itrack), from = min(locus$pos)-2000, to = max(locus$pos)+2000)

gr <- GRanges(seqnames = locus$chr,
              ranges = IRanges(start=locus$pos,end = locus$pos,names = locus$rsid),
              r.squared = locus$r.squared,
              pValue = locus$`-log10(P)`,
              PIP.null = locus$PIP_null,
              PIP.anno = locus$PIP_anno)

dTrack <- DataTrack(gr,
                    data = gr$PIP.anno,
                    name = "PIP (annotation)",
                    fontcolor.title="black",
                    fill="black",
                    cex=1.5,
                    fontfamily="Helvetica",
                    fontsize.title=14,
                    col.axis="black",
                    ylim=c(0,1.0005),
                    pch=20,size=15,
                    background.title="transparent",
                    background.panel="transparent",
                    type="p",
                    col="black")

plotTracks(list(gtrack,itrack,dTrack), from = min(locus$pos)-2000, to = max(locus$pos)+2000)

ucscGenes <- UcscTrack(genome="hg19",
                       table="ncbiRefSeq",
                       track = 'NCBI RefSeq',trackType="GeneRegionTrack",
                       chromosome=chr, rstarts = "exonStarts", rends = "exonEnds",
                       gene = 'name', symbol = 'name', transcript = "name",strand = "strand",
                       stacking = 'pack', showID = TRUE, geneSymbol = TRUE,
                       name="RefSeq Transcripts")

z <- ranges(ucscGenes)

mcols(z)$symbol <- mapIds(org.Hs.eg.db, gsub("\\.[1-9]$", "", mcols(z)$symbol), "SYMBOL","REFSEQ")
ucscGenes2 <- ucscGenes
ranges(ucscGenes2) <- z

displayPars(ucscGenes2) <- list(col=NULL,col.axis="black",background.title="white",fontcolor.title="black",fill="salmon",col.line="lightgrey",fontfamily="Helvetica",fontcolor.group="black",fontsize.group=12,fontsize.title=14,fontface=1)

atrack <- AnnotationTrack(gr,name="Fine-mapped\nSNPs",
                          stacking = "squish",
                          col=NULL,
                          background.title="white",
                          fontcolor.title="black",
                          fontsize.title=14,fill="#2E8B57")


ht <- HighlightTrack(trackList=list(dTrack), start=c(233559500,233791000),width=5000, chromosome=chr,col="transparent",inBackground=T,col="transparent",fill="goldenrod",alpha=0.5)

plotTracks(list(itrack,gtrack,ucscGenes2,atrack,ht),from = min(locus$pos)-2000, to = max(locus$pos)+2000,transcriptAnnotation = "symbol",sizes = NULL,extend.right = 0.05,extend.left = 0.05,main=paste0(unique(locus$lead.snp)," locus"))

dev.off()
pdf(file = "rs4144797-gviz.pdf",width = 7,height = 6)
plotTracks(list(itrack,gtrack,ucscGenes2,atrack,ht),from = min(locus$pos)-2000, to = max(locus$pos)+2000,transcriptAnnotation = "symbol",sizes = NULL,extend.right = 0.05,extend.left = 0.05,main=paste0(unique(locus$lead.snp)," locus"))
dev.off()

```
